src/types.ts

```diff
@@ -137,6 +137,7 @@ export interface RetrievalParams {
   results?: Document[] | string[];
   metadata?: {
     engine?: string;
+    [key: string]: any;
```

Comment:
Use `string | number | boolean` union instead of `any` here, assuming we only want primitives as values and not, for example, callback functions.

Reply:
Makes sense but I was getting other type errors from the engine key here; will leave it to you!